dojo/tools/fortify/parser.py

```diff
@@ -62,26 +62,36 @@ def __init__(self, filename, test):
 
         # All issues obtained, create a map for reference
         issue_map = {}
-        for issue in issues:
-            details = {
-                "Category": issue.find("Category").text,
-                "Folder": issue.find("Folder").text, "Kingdom": issue.find("Kingdom").text,
-                "Abstract": issue.find("Abstract").text,
-                "Friority": issue.find("Friority").text,
-                "FileName": issue.find("Primary").find("FileName").text,
-                "FilePath": issue.find("Primary").find("FilePath").text,
-                "LineStart": issue.find("Primary").find("LineStart").text,
-                "Snippet": issue.find("Primary").find("Snippet").text}
-
-            if issue.find("Source"):
-                source = {
-                    "FileName": issue.find("Source").find("FileName").text,
-                    "FilePath": issue.find("Source").find("FilePath").text,
-                    "LineStart": issue.find("Source").find("LineStart").text,
-                    "Snippet": issue.find("Source").find("Snippet").text}
-                details["Source"] = source
-
-            issue_map.update({issue.attrib['iid']: details})
+        issue_id = "N/A"
+        try:
+            for issue in issues:
+                issue_id = issue.attrib['iid']
+                details = {
+                    "Category": issue.find("Category").text,
+                    "Folder": issue.find("Folder").text, "Kingdom": issue.find("Kingdom").text,
+                    "Abstract": issue.find("Abstract").text,
+                    "Friority": issue.find("Friority").text,
+                    "FileName": issue.find("Primary").find("FileName").text,
+                    "FilePath": issue.find("Primary").find("FilePath").text,
+                    "LineStart": issue.find("Primary").find("LineStart").text}
+
+                if issue.find("Primary").find("Snippet"):
+                    details["Snippet"] = issue.find("Primary").find("Snippet").text
+                else:
+                    details["Snippet"] = "n/a"
+
+                if issue.find("Source"):
+                    source = {
+                        "FileName": issue.find("Source").find("FileName").text,
+                        "FilePath": issue.find("Source").find("FilePath").text,
+                        "LineStart": issue.find("Source").find("LineStart").text,
+                        "Snippet": issue.find("Source").find("Snippet").text}
+                    details["Source"] = source
+
+                issue_map.update({issue.attrib['iid']: details})
+        except AttributeError:
+            print("XML Parsing blew up on issue number:", issue_id)
```

Comment:
can you use logger.warn or similar?

Reply:
I think flake-8-your-pr just answered that question:

```
dojo/tools/fortify/parser.py#L13
expected 2 blank lines, found 1 (E302)
```

Reply:
thanks. final request to make the message a bit more professional...